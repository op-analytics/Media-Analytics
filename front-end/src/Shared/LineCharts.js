import React from 'react';
import { Line } from 'react-chartjs-2';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  chartContainer: {
    width: '100%',
    flex: '0 1 auto',
    maxWidth: '1000px',
  },
}));

function LineCharts({ chartDatasets }) {
  const classes = useStyles();

  return (
    <>
      {chartDatasets.map(data => (
        <div className={classes.chartContainer}>
          <Line
            key={data.datasets[0].data[0].word}
            data={data}
            options={{
              maintainAspectRatio: true,
              responsive: true,
              tooltips: {
                callbacks: {
                  title: (tooltipItem, wordData) => {
                    const response = wordData.datasets[0].data[tooltipItem[0].index];
                    return `${response.year} - ${response.word}`;
                  },
                  label: (toolTipItems => (tooltipItem, wordData) =>
                    toolTipItems.reduce(
                      (accum, { key, value }) =>
                        accum.concat(key + wordData.datasets[0].data[tooltipItem.index][value]),
                      [],
                    ))([
                    {
                      key: 'Frequency: ',
                      value: 'y',
                    },
                    {
                      key: 'Count: ',
                      value: 'wordCount',
                    },
                    {
                      key: 'Rank: ',
                      value: 'rank',
                    },
                  ]),
                },
              },
            }}
          />
        </div>
      ))}
    </>
  );
}

LineCharts.propTypes = {
  chartDatasets: PropTypes.arrayOf(
    PropTypes.shape({
      labels: PropTypes.arrayOf(PropTypes.string),
      datasets: PropTypes.arrayOf(PropTypes.shape({ data: PropTypes.array.isRequired })),
    }),
  ).isRequired,
};

export default LineCharts;
