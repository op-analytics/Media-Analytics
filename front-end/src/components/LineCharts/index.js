import Grid from '@material-ui/core/Grid';
import { makeStyles, ThemeProvider } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';
import {
  byOutletDataset,
  byWordDataset,
  createLegendPayload,
  createTooltip,
  CustomizedDot,
  multipleDatasets,
  singleDataset,
  stringToColour,
} from './utils';

const useStyles = makeStyles(() => ({
  chartContainer: {
    width: '100%',
    height: '50vh',
    flex: '0 1 auto',
    paddingBottom: '6vh',
    paddingTop: '2rem',
  },
  chartTitle: {
    textAlign: 'center',
    '&:first-letter': {
      textTransform: 'uppercase',
    },
  },
  tooltip: {
    margin: 0,
    lineHeight: '24px',
    border: '1px solid #f5f5f5',
    backgroundColor: 'hsla(0,0%,100%,.8)',
    padding: '10px',
  },
  tooltipLabel: {
    color: '#333',
  },
  tooltipLabelFirstWord: {
    color: '#777',
    '&:first-letter': {
      textTransform: 'uppercase',
    },
  },
}));

/**
 * LineCharts A helper component for rendering multiple linecharts
 *
 * @component
 */
function LineCharts({
  datasets,
  xAxisKey,
  yAxisKey,
  displayAbsolute,
  words,
  mediaOutlets,
  allMediaOutlets,
  yearFrom,
  yearTo,
  displayOption,
}) {
  const classes = useStyles();
  const displayed = [];
  let processedData = {};
  let displayAbsoluteLargestValue = 0;

  switch (displayOption) {
    case 'multiple':
      processedData = multipleDatasets(datasets, allMediaOutlets);
      break;
    case 'single':
      processedData = singleDataset(datasets);
      break;
    case 'byOutlet':
      processedData = byOutletDataset(datasets, allMediaOutlets);
      break;
    case 'byWord':
      processedData = byWordDataset(datasets);
      break;
    default:
      processedData = datasets;
  }
  if (displayAbsolute) {
    words.forEach(word => {
      mediaOutlets.forEach(mediaOutlet => {
        const data = processedData.find(obj =>
          obj.data.find(objData =>
            Object.keys(objData).find(key => key.includes(mediaOutlet + word)),
          ),
        );
        if (data) {
          data.data.forEach(dataObj => {
            const currentData = dataObj[mediaOutlet + word + yAxisKey];
            if (currentData > displayAbsoluteLargestValue) {
              displayAbsoluteLargestValue = currentData;
            }
          });
        }
      });
    });
  }

  return (
    <Grid container spacing={1} justify="center">
      {words.map(word => (
        <Grid key={word} item container xs={12} spacing={2} justify="center">
          {mediaOutlets.map(mediaOutlet => {
            const data = processedData.find(obj =>
              obj.data.find(objData =>
                Object.keys(objData).find(key =>
                  key.includes(mediaOutlet + word),
                ),
              ),
            );

            if (data && !displayed.includes(data.title)) {
              displayed.push(data.title);
              return (
                <Grid
                  key={word + mediaOutlet}
                  item
                  xs={
                    displayOption === 'multiple' ? 12 / mediaOutlets.length : 12
                  }
                >
                  <div className={classes.chartContainer} key={data.title}>
                    <h3 className={classes.chartTitle}>{data.title}</h3>
                    <ResponsiveContainer>
                      <LineChart
                        className={classes.chart}
                        data={data.data}
                        margin={{
                          top: 10,
                          right: 30,
                          left: 0,
                          bottom: 0,
                        }}
                      >
                        <CartesianGrid strokeDasharray="3 3" />
                        <XAxis
                          type="number"
                          domain={[yearFrom, yearTo]}
                          dataKey={xAxisKey}
                          tickCount={Math.abs(yearTo - yearFrom)}
                          allowDataOverflow={true} // Forces displayed data to match domain.
                        />
                        <YAxis
                          domain={
                            displayAbsolute
                              ? [0, displayAbsoluteLargestValue]
                              : ['auto', 'auto']
                          }
                        />
                        <Legend
                          payload={createLegendPayload(
                            data,
                            words,
                            mediaOutlets,
                            yAxisKey,
                            allMediaOutlets,
                            displayOption,
                          )}
                        />
                        <Tooltip
                          itemSorter={item1 => item1.value * -1}
                          content={createTooltip(
                            classes,
                            words,
                            mediaOutlets,
                            displayOption,
                            yAxisKey,
                            allMediaOutlets,
                          )}
                        />

                        {words.map(lineWord =>
                          mediaOutlets.map((lineMediaOutlet, index) => {
                            return (
                              <Line
                                key={lineMediaOutlet + lineWord}
                                type="monotone"
                                name={`${
                                  allMediaOutlets.find(
                                    obj => obj.value === lineMediaOutlet,
                                  ).name
                                } - ${lineWord}`}
                                dataKey={lineMediaOutlet + lineWord + yAxisKey}
                                stroke={
                                  displayOption === 'byWord'
                                    ? stringToColour(lineMediaOutlet)
                                    : stringToColour(lineWord)
                                }
                                fill={
                                  displayOption === 'byWord'
                                    ? stringToColour(lineMediaOutlet)
                                    : stringToColour(lineWord)
                                }
                                connectNulls
                                strokeWidth={3}
                                dot={
                                  <CustomizedDot
                                    number={
                                      displayOption === 'byWord' ? 0 : index
                                    }
                                  />
                                }
                                activeDot={{
                                  stroke:
                                    displayOption === 'byWord'
                                      ? stringToColour(lineMediaOutlet)
                                      : stringToColour(lineWord),
                                  strokeWidth: 7,
                                  border: 'white',
                                }}
                              />
                            );
                          }),
                        )}
                      </LineChart>
                    </ResponsiveContainer>
                  </div>
                </Grid>
              );
            }
            return null;
          })}
        </Grid>
      ))}
    </Grid>
  );
}

LineCharts.defaultProps = {
  xAxisKey: 'X',
};

LineCharts.propTypes = {
  /** The datasets to render line charts for */
  datasets: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      data: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
    }),
  ).isRequired,
  /** The key for the x axis of the datasets */
  xAxisKey: PropTypes.string,
  /** The key for the y axis of the datasets */
  yAxisKey: PropTypes.string.isRequired,
  displayAbsolute: PropTypes.bool.isRequired,
  words: PropTypes.arrayOf(PropTypes.string).isRequired,
  mediaOutlets: PropTypes.arrayOf(PropTypes.string).isRequired,
  allMediaOutlets: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
  ).isRequired,
  yearFrom: PropTypes.number.isRequired,
  yearTo: PropTypes.number.isRequired,
  displayOption: PropTypes.string.isRequired,
};

export default LineCharts;