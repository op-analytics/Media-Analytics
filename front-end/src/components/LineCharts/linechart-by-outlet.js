import Grid from '@material-ui/core/Grid';
import PropTypes, { arrayOf, number, shape, string } from 'prop-types';
import React from 'react';
import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';

import { useSharedStyles } from './hooks/useStyles';
import {
  createLegendPayload,
  createTooltip,
  CustomizedDot,
  stringToColour,
} from './utils';

function LinechartByOutlet({ datasets, formParameters, mediaOutlets }) {
  const sharedClasses = useSharedStyles();
  const { words } = formParameters;
  const { outlets } = formParameters;
  const { yearFrom } = formParameters;
  const { yearTo } = formParameters;
  const { yAxisKey } = formParameters;
  const { displayNormalised } = formParameters;
  const { displayOption } = formParameters;

  return (
    <Grid key="container" item container xs={12} spacing={2} justify="center">
      {datasets.map(dataset => (
        <Grid
          key={dataset.title}
          item
          lg={12}
          xs={12}
          className={sharedClasses.gridItemChart}
        >
          <div className={sharedClasses.chartContainer} key={dataset.title}>
            <h3 className={sharedClasses.chartTitle}>{dataset.title}</h3>
            <ResponsiveContainer>
              <LineChart
                data={dataset.data}
                margin={{
                  top: 10,
                  right: 30,
                  left: 0,
                  bottom: 5,
                }}
              >
                <CartesianGrid strokeDasharray="3 3" />
                <XAxis
                  type="number"
                  domain={[yearFrom, yearTo]}
                  dataKey="year"
                  tickCount={Math.abs(yearTo - yearFrom)}
                  allowDataOverflow // Forces displayed data to match domain.
                />
                <YAxis
                  domain={displayNormalised ? [0, 1] : ['auto', 'auto']}
                  width={75}
                  allowDataOverflow // Forces displayed data to match domain.
                />
                <Legend
                  payload={createLegendPayload(
                    dataset.data,
                    words,
                    outlets,
                    yAxisKey,
                    mediaOutlets,
                    displayOption,
                  )}
                />
                <Tooltip
                  itemSorter={item1 => item1.value * -1}
                  content={createTooltip(
                    sharedClasses,
                    words,
                    outlets,
                    displayOption,
                    yAxisKey,
                    mediaOutlets,
                    datasets,
                  )}
                />

                {words.map(word => {
                  return outlets.map(outlet => (
                    <Line
                      key={word + outlet}
                      type="monotone"
                      dataKey={outlet + word + yAxisKey}
                      stroke={stringToColour(word)}
                      fill={stringToColour(word)}
                      connectNulls
                      strokeWidth={3}
                      dot={<CustomizedDot number={0} />}
                      activeDot={{
                        stroke: stringToColour(word),
                        strokeWidth: 7,
                        border: 'white',
                      }}
                    />
                  ));
                })}
              </LineChart>
            </ResponsiveContainer>
          </div>
        </Grid>
      ))}
    </Grid>
  );
}

LinechartByOutlet.propTypes = {
  datasets: arrayOf(
    shape({
      title: string.isRequired,
      data: arrayOf(
        shape({
          word: string.isRequired,
          outlet: string.isRequired,
          year: string.isRequired,
          outletWordRank: number,
          outletWordCount: number,
          outletWordFreq: number,
        }).isRequired,
      ).isRequired,
    }).isRequired,
  ).isRequired,
  formParameters: shape({
    words: arrayOf(string).isRequired,
    outlets: arrayOf(string).isRequired,
    yearFrom: number.isRequired,
    yearTo: number.isRequired,
    yAxisKey: string.isRequired,
    displayNormalised: PropTypes.bool.isRequired,
    displayOption: string.isRequired,
  }).isRequired,
  mediaOutlets: arrayOf(
    shape({
      title: string.isRequired,
      value: string.isRequired,
    }),
  ).isRequired,
};

export default LinechartByOutlet;
