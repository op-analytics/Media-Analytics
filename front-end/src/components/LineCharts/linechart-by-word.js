import Grid from '@material-ui/core/Grid';
import React from 'react';
import {
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';
import {
  createLegendPayload,
  createTooltip,
  CustomizedDot,
  stringToColour,
} from './utils';

function LinechartByWord({ datasets, formParameters, mediaOutlets, classes }) {
  const words = formParameters.words;
  const outlets = formParameters.outlets;
  const yearFrom = formParameters.yearFrom;
  const yearTo = formParameters.yearTo;
  const yAxisKey = formParameters.yAxisKey;
  const displayNormalised = formParameters.displayNormalised;
  const displayOption = formParameters.displayOption;

  return (
    <Grid key={'container'} item container xs={12} spacing={2} justify="center">
      {datasets.map(dataset => (
        <Grid
          key={dataset.title}
          item
          lg={12}
          xs={12}
          className={classes.gridItemChart}
        >
          <div className={classes.chartContainer} key={dataset.title}>
            <h3 className={classes.chartTitle}>{dataset.title}</h3>
            <ResponsiveContainer>
              <LineChart
                data={dataset.data}
                margin={{
                  top: 10,
                  right: 30,
                  left: 0,
                  bottom: 5,
                }}
              >
                <CartesianGrid strokeDasharray="3 3" />
                <XAxis
                  type="number"
                  domain={[yearFrom, yearTo]}
                  dataKey={'year'}
                  tickCount={Math.abs(yearTo - yearFrom)}
                  allowDataOverflow={true} // Forces displayed data to match domain.
                />
                <YAxis
                  domain={displayNormalised ? [0, 1] : ['auto', 'auto']}
                  width={75}
                  allowDataOverflow={true} // Forces displayed data to match domain.
                />
                <Legend
                  payload={createLegendPayload(
                    dataset.data,
                    words,
                    outlets,
                    mediaOutlets,
                    displayOption,
                  )}
                />
                <Tooltip
                  itemSorter={item1 => item1.value * -1}
                  content={createTooltip(
                    classes,
                    words,
                    outlets,
                    displayOption,
                    yAxisKey,
                    mediaOutlets,
                    datasets
                  )}
                />

                {outlets.map((outlet) => {
                  return (
                    <Line
                      key={outlet}
                      type="monotone"
                      dataKey={outlet + yAxisKey}
                      stroke={stringToColour(outlet)}
                      fill={stringToColour(outlet)}
                      connectNulls
                      strokeWidth={3}
                      dot={<CustomizedDot number={0} />}
                      activeDot={{
                        stroke: stringToColour(outlet),
                        strokeWidth: 7,
                        border: 'white',
                      }}
                    />
                  );
                })}
              </LineChart>
            </ResponsiveContainer>
          </div>
        </Grid>
      ))}
    </Grid>
  );
}

export default LinechartByWord;
