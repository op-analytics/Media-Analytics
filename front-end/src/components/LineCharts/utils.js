/* eslint-disable react/prop-types */
import React from 'react';

/**
 * A tooltip factory for creating tooltips dynamicaly
 *
 * @param {Object} classes Classes to apply to the tooltip
 * @param {Object[]} items The items to render on the tooltip
 * @returns {Element}
 */
export const createTooltip = (classes, items, dot) => {
  const ToolTip = ({ active, payload, label }) => {
    if (active && dot) {
      return (
        <div className={classes.tooltip}>
          <h3>{label}</h3>
          {items.map(item => {
            let payloadItem = payload[0].payload[dot + item.key];
            if (item.formatFunction)
              payloadItem = item.formatFunction(payloadItem);
            return (
              <p className={classes.tooltipLabel} key={dot + item.title}>
                <span className={classes.tooltipLabelFirstWord}>
                  {`${item.title}: `}
                </span>
                {payloadItem}
              </p>
            );
          })}
        </div>
      );
    }
    return null;
  };
  return ToolTip;
};

/**
 * A deterministic converter of a string to a hexidecimal colour
 *
 * @param {String} str
 * @returns {String}
 */

export const stringToColour = str => {
  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    hash = str.charCodeAt(i) + ((hash << 5) - hash);
  }
  let colour = '#';
  for (let i = 0; i < 3; i++) {
    let value = (hash >> (i * 8)) & 0xff;
    colour += ('00' + value.toString(16)).substr(-2);
  }
  return colour;
};
