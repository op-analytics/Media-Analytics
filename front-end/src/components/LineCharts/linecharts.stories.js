import React from 'react';
import LineCharts from '.';

export default {
  title: 'LineCharts',
  component: LineCharts,
};

const chartDatasetsWithOneWord = [
  {
    title: 'bob',
    data: [{ y: 1, x: 2 }, { y: 0.3, x: 3 }, { y: 0.6, x: 7 }],
    tooltipItems: [{ title: 'y', key: 'y' }],
  },
];

const chartDatasetsWithTwoWords = [
  {
    title: 'bob',
    data: [{ y: 1, x: 2 }, { y: 0.3, x: 3 }, { y: 0.6, x: 7 }],
  },
  {
    title: 'bill',
    data: [{ y: 1, x: 2 }, { y: 0.3, x: 3 }, { y: 0.6, x: 7 }],
  },
];

const getProps = dataSource => ({
  datasets: dataSource,
  yAxisKey: 'y',
  xAxisKey: 'x',
  displayAbsolute: false,
  tooltipItems: [{ title: 'y', key: 'y' }],
});

export const LineChartWithOneDataSource = () => (
  <LineCharts {...getProps(chartDatasetsWithOneWord)} />
);

export const LineChartWithTwoDataSources = () => (
  <LineCharts {...getProps(chartDatasetsWithTwoWords)} />
);
