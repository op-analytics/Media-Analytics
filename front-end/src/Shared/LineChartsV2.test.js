import React from 'react';
import { shallow } from 'enzyme';
import LineCharts from './LineChartsV2';

const chartDatasetsWithOneWord = [{ labels: [], datasets: [{ data: [{ word: 'bob' }] }] }];
const chartDatasetsWithTwoWords = [
  { labels: [], datasets: [{ data: [{ word: 'bob' }] }] },
  { labels: [], datasets: [{ data: [{ word: 'phil' }] }] },
];
// eslint-disable-next-line react/jsx-props-no-spreading
const getWrapper = props => shallow(<LineCharts {...props} />);

describe('LineChart', () => {
  it('renders a chart for a dataset of one word', () => {
    const wrapper = getWrapper({ chartDatasets: chartDatasetsWithOneWord });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a chart for a dataset of multiple words', () => {
    const wrapper = getWrapper({ chartDatasets: chartDatasetsWithTwoWords });
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const wrapper = getWrapper({ chartDatasets: chartDatasetsWithOneWord });
    expect(wrapper).toMatchSnapshot();
  });
});
