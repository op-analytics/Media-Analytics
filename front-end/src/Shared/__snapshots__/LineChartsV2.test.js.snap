// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart renders a chart for a dataset of multiple words 1`] = `
<Fragment>
  <div
    className="makeStyles-chartContainer-1"
    key="undefinedcont"
  >
    <h1
      className="makeStyles-chartTitle-2"
      key="undefinedh1"
    />
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        layout="horizontal"
        margin={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 30,
            "top": 10,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="year"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              classes={
                Object {
                  "chartContainer": "makeStyles-chartContainer-1",
                  "chartTitle": "makeStyles-chartTitle-2",
                  "customTooltip": "makeStyles-customTooltip-3",
                  "customTooltipLabel": "makeStyles-customTooltipLabel-4",
                  "customTooltipLabelSpan": "makeStyles-customTooltipLabelSpan-5",
                }
              }
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Line
          activeDot={
            Object {
              "border": "white",
              "stroke": "#3F51B5",
              "strokeWidth": 7,
            }
          }
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="freq"
          dot={
            Object {
              "strokeWidth": 5,
            }
          }
          fill="#8884d8"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="#8884d8"
          strokeWidth={3}
          type="monotone"
          xAxisId={0}
          yAxisId={0}
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
  <div
    className="makeStyles-chartContainer-1"
    key="undefinedcont"
  >
    <h1
      className="makeStyles-chartTitle-2"
      key="undefinedh1"
    />
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        layout="horizontal"
        margin={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 30,
            "top": 10,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="year"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              classes={
                Object {
                  "chartContainer": "makeStyles-chartContainer-1",
                  "chartTitle": "makeStyles-chartTitle-2",
                  "customTooltip": "makeStyles-customTooltip-3",
                  "customTooltipLabel": "makeStyles-customTooltipLabel-4",
                  "customTooltipLabelSpan": "makeStyles-customTooltipLabelSpan-5",
                }
              }
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Line
          activeDot={
            Object {
              "border": "white",
              "stroke": "#3F51B5",
              "strokeWidth": 7,
            }
          }
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="freq"
          dot={
            Object {
              "strokeWidth": 5,
            }
          }
          fill="#8884d8"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="#8884d8"
          strokeWidth={3}
          type="monotone"
          xAxisId={0}
          yAxisId={0}
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
</Fragment>
`;

exports[`LineChart renders a chart for a dataset of one word 1`] = `
<Fragment>
  <div
    className="makeStyles-chartContainer-1"
    key="undefinedcont"
  >
    <h1
      className="makeStyles-chartTitle-2"
      key="undefinedh1"
    />
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        layout="horizontal"
        margin={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 30,
            "top": 10,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="year"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              classes={
                Object {
                  "chartContainer": "makeStyles-chartContainer-1",
                  "chartTitle": "makeStyles-chartTitle-2",
                  "customTooltip": "makeStyles-customTooltip-3",
                  "customTooltipLabel": "makeStyles-customTooltipLabel-4",
                  "customTooltipLabelSpan": "makeStyles-customTooltipLabelSpan-5",
                }
              }
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Line
          activeDot={
            Object {
              "border": "white",
              "stroke": "#3F51B5",
              "strokeWidth": 7,
            }
          }
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="freq"
          dot={
            Object {
              "strokeWidth": 5,
            }
          }
          fill="#8884d8"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="#8884d8"
          strokeWidth={3}
          type="monotone"
          xAxisId={0}
          yAxisId={0}
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
</Fragment>
`;

exports[`LineChart should render correctly 1`] = `
<Fragment>
  <div
    className="makeStyles-chartContainer-1"
    key="undefinedcont"
  >
    <h1
      className="makeStyles-chartTitle-2"
      key="undefinedh1"
    />
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <LineChart
        barCategoryGap="10%"
        barGap={4}
        layout="horizontal"
        margin={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 30,
            "top": 10,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="year"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              classes={
                Object {
                  "chartContainer": "makeStyles-chartContainer-1",
                  "chartTitle": "makeStyles-chartTitle-2",
                  "customTooltip": "makeStyles-customTooltip-3",
                  "customTooltipLabel": "makeStyles-customTooltipLabel-4",
                  "customTooltipLabelSpan": "makeStyles-customTooltipLabelSpan-5",
                }
              }
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Line
          activeDot={
            Object {
              "border": "white",
              "stroke": "#3F51B5",
              "strokeWidth": 7,
            }
          }
          animateNewValues={true}
          animationBegin={0}
          animationDuration={1500}
          animationEasing="ease"
          connectNulls={false}
          dataKey="freq"
          dot={
            Object {
              "strokeWidth": 5,
            }
          }
          fill="#8884d8"
          hide={false}
          isAnimationActive={true}
          legendType="line"
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          points={Array []}
          stroke="#8884d8"
          strokeWidth={3}
          type="monotone"
          xAxisId={0}
          yAxisId={0}
        />
      </LineChart>
    </ResponsiveContainer>
  </div>
</Fragment>
`;
